File: framework/ui-elements/src/displays/components/DisplayStamp.js
import React from "react";
import {withStyles} from 'alexandria-ui-elements/src/util/muiStylesCompat';
import AbstractDisplayStamp from "../../../gen/displays/components/AbstractDisplayStamp";
import DisplayStampNotifier from "../../../gen/displays/notifiers/DisplayStampNotifier";
import DisplayStampRequester from "../../../gen/displays/requesters/DisplayStampRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';
import {withSnackbar} from "alexandria-ui-elements/src/util/notistackCompat";

const styles = theme => ({});

class DisplayStamp extends AbstractDisplayStamp {

	constructor(props) {
		super(props);
		this.notifier = new DisplayStampNotifier(this);
		this.requester = new DisplayStampRequester(this);
        this.state = {
            ...this.state,
            display : null
        };
	};

	render() {
		if (!this.state.visible) return (<React.Fragment/>);
		return (<div style={this.style()}>{this.renderInstances(null, null, this.style())}</div>);
	};

	refreshDisplay = () => {
	};
}

export default withStyles(styles, { withTheme: true })(withSnackbar(DisplayStamp));
DisplayFactory.register("DisplayStamp", withStyles(styles, { withTheme: true })(withSnackbar(DisplayStamp)));

File: framework/ui/src/io/intino/alexandria/ui/displays/components/DisplayStamp.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.Display;
import io.intino.alexandria.ui.displays.notifiers.DisplayStampNotifier;

import java.util.UUID;

public class DisplayStamp<DN extends DisplayStampNotifier, B extends Box> extends AbstractDisplayStamp<B> {
    private Display display;

    public DisplayStamp(B box) {
        super(box);
    }

    @Override
    public void didMount() {
        super.didMount();
        add(this.display);
        refresh();
    }

    public <D extends Display> D get() {
        return (D) this.display;
    }

    public <D extends Display> D display() {
        return get();
    }

    public void display(Display display) {
        this.display = display;
        this.display.id(UUID.randomUUID().toString());
    }

    @Override
    public void refresh() {
        super.refresh();
        if (display != null) display.refresh();
    }
}